Add market version dropdown

md0x opened the pull request:
Signed-off-by: Pablo Maldonado <pablo@oceanprotocol.com>

- addresses #1450

Changes proposed in this PR:

- Add market v3/v4 dropdown 

<img width="510" alt="Screen Shot 2022-05-23 at 16 56 09" src="https://user-images.githubusercontent.com/90316/169859468-922b5ffc-290e-4bcc-b633-395610779624.png">

<img width="571" alt="Screen Shot 2022-05-23 at 16 55 32" src="https://user-images.githubusercontent.com/90316/169859411-0c5049ac-8b12-4196-97ba-38af7bb02249.png">

mihaisc commented:
@kremalicious any idea why the vercel build are mixed up?

kremalicious commented:
> any idea why the vercel build are mixed up?

it's complex, explained in Slack

md0x commented:
> Think we should go a bit more minimal with that as users won't constantly interact with this dropdown, so it should not be on same visual hierarchy as all the other menubar actions.
> 
> So something like this:
> 
> <img alt="Screen Shot 2022-03-16 at 12 38 21" width="523" src="https://user-images.githubusercontent.com/90316/158591621-2b07e4fb-c5b6-414c-815c-26c5608db6da.png">

@kremalicious I have updated the dropdown design and updated the screenshots accordingly in the PR description. Was the offset of the dropdown intended? 
![image](https://user-images.githubusercontent.com/53186719/158645415-ef500d3e-b662-4261-b3b6-150d4fdc1f76.png)

jamiehewitt15 commented:
Just want to confirm, @kremalicious are you taking this one now?

kremalicious commented on src/components/molecules/Menu.tsx:
```diff
@@ -9,6 +9,13 @@ import Badge from '../atoms/Badge'
+import appConfig from '../../../app.config'
+
+const redirectToMarketV4 = (): void => {
+  window.location.href = appConfig.marketV4Url
```
TODO: just use a plain link with `href`

kremalicious commented on src/components/molecules/Menu.tsx:
```diff
@@ -39,10 +46,34 @@ export default function Menu(): ReactElement {
       </Link>
+      <Tooltip
+        content={
+          <ul className={styles.preferencesDetails}>
```
TODO: way too much markup for what we want to do, simplify

kremalicious commented:
TODO: there is no version badge on mobile, so also no way to switch versions on mobile

mihaisc commented:
instead of a dropdown wouldn't it better to have a switch?
https://codepen.io/alvarotrigo/pen/wvPRrZW
![toggle](https://user-images.githubusercontent.com/26000280/169848322-6ed78c29-c2ce-4c08-ae5d-0e04f29c944e.gif)

kremalicious commented:
> instead of a dropdown wouldn't it better to have a switch?

was also thinking that but our transition between both versions is not so smooth as we are switching domains. So switch then feels a bit weird as user would never really see it move

kremalicious commented:
ok, currently pushed state works because menu only has 1 item left. For the v4 versions we need to figure out some other placement in small viewport view.

Regardless, we have to touch this again as we have all those different market URLs which will then change and switch in the future, so every time we do that, we have to update future `v3` branch always together with future `main` branch

mihaisc commented:
Ah, didn't think this through
